### PR TITLE
Fix MSVC tests + add mingw support

### PR DESCRIPTION
**Pull request by madebr**
This PR fixes various problems with buiding dethrace with mingw and MSVC.
It also enables running the Windows tests on github actions.

I've documented the reasoning behind each commit in the commit messages.

- This is the result of github actions: https://github.com/madebr/dethrace/actions/runs/1633319741
- I've sent the cglm patch to upstream: https://github.com/recp/cglm/pull/234

### CODE REVIEW

File: test/BRSRC13/test_resource.c [hunk 1]
```diff
@@ -62,17 +62,21 @@ void test_resource_BrResFree_Child() {
     header = (resource_header*)(((char*)child) - sizeof(resource_header));
     TEST_ASSERT_EQUAL_INT(0xDEADBEEF, header->magic_num);
     BrResFree(r);
+
+#ifndef _DEBUG
```
Comment (dethrace-labs) [inline test/BRSRC13/test_resource.c:66]: why do we need to add this guard?

File: test/DETHRACE/test_controls.c [hunk 2]
```diff
@@ -24,7 +24,7 @@ void test_controls_CheckKevKeys() {
         gKeys_pressed = 0;
         result = KevKeyService();
     }
-    sleep(2);
+    sleep_s(2);
```
Comment (dethrace-labs) [inline test/DETHRACE/test_controls.c:27]: what is `sleep_s`?

Comment (dethrace-labs) [reply test/DETHRACE/test_controls.c:27]: nevermind, I see

File: test/BRSRC13/test_resource.c (hunk 1 above)
Comment (madebr) [reply test/BRSRC13/test_resource.c:66]: MSVC in debug mode overwrites all free'd memory with some mask to detect use-after-free.
`BrResFree(r)` will also free all its children, so we cannot access the `header` pointer anymore.

**Review by dethrace-labs — APPROVED** [review]

File: test/BRSRC13/test_resource.c (hunk 1 above)
Comment (dethrace-labs) [reply test/BRSRC13/test_resource.c:66]: ah, got it

**Comment by dethrace-labs** [comment]
thanks for this! I didn't know about `UnitySetTestFile` - I've been ignoring the reported file name in test errors because it was always reporting `main.c` :)

**Pull request merged**